oak_containers_stage1/src/dice.rs

```diff
@@ -44,8 +68,111 @@ const MEMMAP_PATH: &str = "/sys/firmware/memmap";
 /// The path for reading the physical memory from the mem device.
 const PHYS_MEM_PATH: &str = "/dev/mem";
 
+/// Builds the DICE evidence and certificate authority for the next DICE layer.
+pub struct DiceDataBuilder {
+    evidence: Evidence,
+    signing_key: SigningKey,
+    system_image_measurement: Vec<u8>,
+}
+
+impl DiceDataBuilder {
```

Comment:
I think originally we were planning to have some of this logic in the `oak_dice` crate.

I.e. to have something like:
```rust
struct Attester {
  fn create(DiceData dice_data) -> Self;
  fn generate_evidence(Measurements measurements, ApplicationKeys application_keys) -> Evidence;
  fn generate_dice_data() -> DiceData;
}
```
But it could be a separate PR

Reply:
Yes, this requires some further design. I started with this approach, but since `oak_dice` is used by Stage 0 and we are trying to avoid protos as a dependency in Stage 0 adding this there is problematic. We would either have to create two dice crates (one with protos and one without) or accept that we will have protobufs as a dependency in Stage 0 and the Restricted Kernel.

Reply:
The crate with protos could be the `oak_remote_attestation`, which will depend on the `oak_dice`